### ISSUE TITLE
Facial-recognition

### ISSUE THREAD

**Issue by Machinelearning1954** [Machinelearning1954/Facial-recognition-#3]
#!/usr/bin/env python3

"""Loads the Labeled Faces in the Wild (LFW) dataset using scikit-learn."""

from sklearn.datasets import fetch_lfw_people
import os
import joblib
Research Findings: NCIS Stalking/Harassment Cases
Case Example: Navy Hospital Employee Stalking Military Service Member (April 2023)
Source: U.S. Department of Justice Press Release (https://www.justice.gov/usao-sdca/pr/navy-hospital-employee-pleads-guilty-stalking-military-service-member)
Summary: A former contract employee at a San Diego Navy hospital (Jonathan Sandoval) pleaded guilty to stalking a military service member who was his former co-worker.
Actions: The stalking included unwanted acts like rifling through the victim's belongings, displaying a knife, sending unwanted messages, installing a hidden camera in the victim's office, recording the victim undressing, and emailing nude images with threats.
Legal Outcome: The employee pleaded guilty to federal stalking charges (Title 18, U.S.C., Section 2261A(2)(B)), facing a maximum penalty of five years in prison and a $250,000 fine.
NCIS Involvement: The Naval Criminal Investigative Service (NCIS) investigated the case. The Special Agent in Charge emphasized NCIS's commitment to investigating such criminality.
Reporting: The press release advises reporting cyberstalking by active duty service members to specific tip lines for Navy/Marines (NCIS), Army (CID), and Air Force (OSI), or the DoD Hotline. Stalking by non-active-duty members should be reported to local law enforcement or the FBI.
Other Mentions (from initial search):
Several search results mentioned Pauley Perrette, a former actress on the show "NCIS," and her personal experiences with stalking and advocacy for changes in stalking laws. While related to the show's name, these are about the actress personally, not the agency's cases.
Another case involved an Everett, Washington man convicted of cyberstalking a former partner, highlighting that stalking, including cyberstalking, is a federal crime investigated by agencies like the FBI (mentioned in the DOJ release for non-active-duty cases).
Key Takeaways:
NCIS investigates stalking and harassment cases involving Navy and Marine Corps personnel.
Stalking, including cyberstalking and using hidden cameras, is treated as a serious federal crime.
There are specific channels within the military (NCIS, CID, OSI tip lines, DoD Hotline) for reporting stalking by service members.
Stalking by civilians (even if the victim is military) may involve local law enforcement or the FBI.

# Define where to cache the dataset
# Using the default scikit-learn data home is usually fine,
# but specifying it explicitly for clarity within the project.
# Note: fetch_lfw_people uses joblib which caches by default in ~/scikit_learn_data
# We'll let it use the default cache location for simplicity.

def load_lfw_dataset(min_faces=70, resize=0.4):
    """Loads the LFW dataset, downloading if necessary.

    Args:
        min_faces (int): Minimum number of faces per person to include.
        resize (float): Ratio to resize the images.

    Returns:
        Bunch: The LFW dataset object from scikit-learn.
    """
    print(f"Loading LFW dataset (min_faces_per_person={min_faces}, resize={resize})...")
    print("This may take several minutes the first time as the data is downloaded (~200MB) and processed.")
    
    # Fetch the dataset. It will be downloaded and cached automatically
    # in the user's home directory (e.g., ~/scikit_learn_data/lfw_home/)
    lfw_people = fetch_lfw_people(min_faces_per_person=min_faces, resize=resize)
    
    n_samples, h, w = lfw_people.images.shape
    n_features = lfw_people.data.shape[1]
    n_classes = lfw_people.target_names.shape[0]

    print("\nDataset loaded successfully.")
    print(f"Total dataset size:")
    print(f"  Number of samples: {n_samples}")
    print(f"  Number of features (pixels): {n_features}")
    print(f"  Number of classes (people): {n_classes}")
    print(f"  Image dimensions: {h}x{w}")
    
    return lfw_people

if __name__ == "__main__":
    # Example usage: Load the dataset with default parameters
    dataset = load_lfw_dataset()
    print("\nFirst 5 target names (people):")
    print(dataset.target_names[:5])
    print("\nFirst 5 target labels (indices):")
    print(dataset.target[:5])
